src/main/scala/URAlgorithm.scala

```diff
@@ -122,469 +139,555 @@ case class URAlgorithmParams(
+        case (item, (None, Some(rankPropMap)))                => item -> rankPropMap
+        case (item, _)                                        => item -> DataMap()
       }
-    } else data.fieldsRDD
+    } else {
+      sc.emptyRDD
+    }
 
     logger.info("Correlators created now putting into URModel")
-    val model = new URModel(
-      Some(cooccurrenceCorrelators),
-      Some(allPropertiesRDD),
-      ap.indexName,
-      dateNames,
-      typeMappings = Some(nonDefaultMappings))
-    model.save(ap)
+    new URModel(
+      coocurrenceMatrices = cooccurrenceCorrelators,
+      fieldsRDD = allPropertiesRDD,
+      typeMappings = getRankMapping(backfillFieldNames),
+      propertiesRDD = sc.emptyRDD).save(dateNames, esIndex, esType)
     new NullModel
   }
 
   /** This function creates a URModel from an existing index in Elasticsearch + new popularity ranking
-    * It is used when you want to re-calc the popularity model between training on useage data. It leaves
-    * the part of the model created from usage data alone and only modifies the popularity ranking.
-    */
-  def calcPop(
-    sc: SparkContext,
-    data: PreparedData,
-    dateNames: Option[List[String]] = None,
-    backfillFieldName: String = ""): NullModel = {
-
-    val backfillParams = ap.backfillField.getOrElse(defaultURAlgorithmParams.DefaultBackfillParams)
-    val backfillEvents = backfillParams.eventNames.getOrElse(List(ap.eventNames.head))//default to first/primary event
-    val durationString = ap.backfillField.getOrElse(defaultURAlgorithmParams.DefaultBackfillParams).duration
-        .getOrElse(defaultURAlgorithmParams.DefaultBackfillDuration)
-    val duration = Duration(durationString).toSeconds.toInt
-    val end = ap.backfillField.getOrElse(defaultURAlgorithmParams.DefaultBackfillParams).offsetDate
-    val popModel = PopModel.calc(
-      Some(backfillParams.backfillType.getOrElse(defaultURAlgorithmParams.DefaultBackfillType)),
-      backfillEvents,
-      ap.appName,
-      duration,
-      end)(sc)
-    val popRDD = if (popModel.nonEmpty) {
-      val model = popModel.get.map { case (item, rank)  =>
-        val newPM = Map(backfillFieldName -> JDouble(rank))
-        (item, PropertyMap(newPM, DateTime.now, DateTime.now))
-      }
-      Some(model)
-    } else None
-
-    val propertiesRDD = if (popModel.nonEmpty) {
-      val currentMetadata = EsClient.getRDD(sc, ap.indexName, ap.typeName)
-      if (currentMetadata.nonEmpty) { // may be an empty index so ignore
-        Some(popModel.get.cogroup[collection.Map[String, AnyRef]](currentMetadata.get)
-          .map { case (item, (ranks, pms)) =>
-            if (ranks.nonEmpty) pms.head + (backfillFieldName -> ranks.head)
-            else if (pms.nonEmpty) pms.head
-            else Map.empty[String, AnyRef] // could happen if only calculating popularity, which may leave out items with
-            // no events
-          })
-      } else None
-    } else None
```

Comment:
There may be a bug here if you only want to calculate the popModel, but the intended use is to update the popModel without requiring the entire model to be recalculated. For instance the popModel is calculated every day, and the recs model every week. 

Good point though, some may want only popularity.